alerting/src/main/kotlin/org/opensearch/alerting/util/DocLevelMonitorQueries.kt

```diff
@@ -202,43 +239,152 @@ class DocLevelMonitorQueries(private val client: Client, private val clusterServ
+        updateMappingRequest.source(mapOf<String, Any>("properties" to updatedProperties))
+        var updateMappingResponse = AcknowledgedResponse(false)
+        try {
+            updateMappingResponse = client.suspendUntil {
+                client.admin().indices().putMapping(updateMappingRequest, it)
+            }
+            return Pair(updateMappingResponse, targetQueryIndex)
+        } catch (e: Exception) {
+            // If we reached limit for total number of fields in mappings, do a rollover here
+            if (e.message?.contains("Limit of total fields") == true) {
+                targetQueryIndex = rolloverQueryIndex(monitor)
+                try {
+                    // PUT mappings to newly created index
+                    val updateMappingRequest = PutMappingRequest(targetQueryIndex)
                     updateMappingRequest.source(mapOf<String, Any>("properties" to updatedProperties))
-                    val updateMappingResponse: AcknowledgedResponse = client.suspendUntil {
+                    updateMappingResponse = client.suspendUntil {
                         client.admin().indices().putMapping(updateMappingRequest, it)
                     }
-
-                    if (updateMappingResponse.isAcknowledged) {
-                        val indexRequests = mutableListOf<IndexRequest>()
-                        queries.forEach {
-                            var query = it.query
-                            flattenPaths.forEach { fieldPath ->
-                                query = query.replace("$fieldPath:", "${fieldPath}_${indexName}_$monitorId:")
-                            }
-                            val indexRequest = IndexRequest(queryIndex)
-                                .id(it.id + "_${indexName}_$monitorId")
-                                .source(
-                                    mapOf(
-                                        "query" to mapOf("query_string" to mapOf("query" to query)),
-                                        "monitor_id" to monitorId,
-                                        "index" to indexName
-                                    )
-                                )
-                            indexRequests.add(indexRequest)
-                        }
-                        if (indexRequests.isNotEmpty()) {
-                            val bulkResponse: BulkResponse = client.suspendUntil {
-                                client.bulk(
-                                    BulkRequest().setRefreshPolicy(refreshPolicy).timeout(indexTimeout).add(indexRequests), it
-                                )
-                            }
-                        }
+                } catch (e: Exception) {
+                    // If we reached limit for total number of fields in mappings after rollover
+                    // it means that source index has more then (FIELD_LIMIT - 3) fields (every query index has 3 fields defined)
```

Comment:
Can't we rollover multiple times?

Reply:
queryIndex already has 3 fields in mappings so when source_index could have 999 fields and we couldn't succeed ever so we must abort

Reply:
but in any other case, yea, we can rollover

Reply:
Ahh I see. Can we add a TODO to support indices that have at least 997 fields by looking into breaking up the data amongst multiple rolled over indices?

Reply:
done!